vmtkScripts/contrib/vmtknumpytocenterlines.py

```diff
@@ -0,0 +1,111 @@
+## Module:    $RCSfile: vmtknumpytocenterlines.py,v $
+## Language:  Python
+## Date:      June 10, 2017
+## Version:   1.4
+
+##   Copyright (c) Richard Izzo, Luca Antiga, David Steinman. All rights reserved.
+##   See LICENCE file for details.
+
+##      This software is distributed WITHOUT ANY WARRANTY; without even
+##      the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+##      PURPOSE.  See the above copyright notices for more information.
+
+## Note: this class was contributed by
+##       Richard Izzo (Github @rlizzo)
+##       University at Buffalo
+##       The Jacobs Institute
+
+from __future__ import absolute_import  # NEEDS TO STAY AS TOP LEVEL MODULE FOR Py2-3 COMPATIBILITY
+import vtk
+import sys
+
+from vmtk import vtkvmtk
+from vmtk import vmtkrenderer
+from vmtk import pypes
+
+try:
+    import numpy as np
+except ImportError:
+    raise ImportError('Unable to Import vmtknumpytosurface module, numpy is not installed')
+
+class vmtkNumpyToCenterlines(pypes.pypeScript):
+    def __init__(self):
+
+        pypes.pypeScript.__init__(self)
+
+        self.ArrayDict = None
+
+        self.SetScriptName('vmtkNumpyToCenterlines')
+        self.SetScriptDoc('Takes a nested python dictionary containg numpy arrays specifying vertex '
+                          'points, associated scalar data, and cell data specifying line connectivity'
+                          'and returns a VTK centerlines vtkPolyData file')
+        self.SetInputMembers([
+            ['ArrayDict', 'i', 'dict', 1, '', 'the input dictionary']])
+        self.SetOutputMembers([
+            ['Centerlines', 'o', 'vtkPolyData', 1, '', 'the output surface', 'vmtksurfacewriter']])
+
+    def Execute(self):
+
+        polyData = vtk.vtkPolyData()
+
+        points = vtk.vtkPoints()
+        for xyzPoint in self.ArrayDict['Points']:
+            points.InsertNextPoint(xyzPoint)
+
+        polyData.SetPoints(points)
+
+        self.PrintLog('converting numpy array to centerlines')
+        pointDataKeys = self.ArrayDict['PointData'].keys()
+        for key in pointDataKeys:
+
+            pointDataDType = str(self.ArrayDict['PointData'][key].dtype)
+            if pointDataDType.find('float') != -1:
```

Comment:
Here (and in all the other relevant spots) you could check the `dtype` in a more idiomatic way: https://stackoverflow.com/questions/22471644/how-do-i-check-if-a-numpy-dtype-is-integral